### PR TITLE
Improving the output friction energy : change documentation

### PR DESCRIPTION
**Pull request by MahaHachani**
#### Description of the feature or the bug

Improving the output friction energy : change documentation

#### Description of the changes

Improving the output friction energy : change documentation
/H3D/NODA/CSE_FRICG to /H3D/NODA/CSE_FRIC for global energy friction
/H3D/NODA/CSE_FRIC/INTER
Adapt the H3d noda scalar to output by interface, add description of the different output by interface

### CODE REVIEW

File: engine/source/output/h3d/h3d_build_cpp/c_h3d_create_nodal_scalar_datatype.cpp [hunk 1]
```diff
@@ -69,20 +69,25 @@ extern "C"
...
     unsigned int elem_count = 1;
     char* edata_type;
     
     char * CH_INTER_ID = new char [100];
-    CH_INTER_ID[0] ='\0'; 
+    CH_INTER_ID[0] ='\0';  
 
-    cname_len = *size + 10;
+    chinter_len = 10 + *size_inter0;
+    chinter=(char*) malloc(sizeof(char)*chinter_len);
```
Comment (laurent-altr) [inline engine/source/output/h3d/h3d_build_cpp/c_h3d_create_nodal_scalar_datatype.cpp:86]: Memory leak. Not freed.

**Review by laurent-altr — CHANGES_REQUESTED** [review]
c_h3d_create_nodal_scalar_datatype.cpp :

File: engine/source/output/h3d/h3d_build_cpp/c_h3d_create_nodal_scalar_datatype.cpp (hunk 1 above)
Comment (MahaHachani) [reply engine/source/output/h3d/h3d_build_cpp/c_h3d_create_nodal_scalar_datatype.cpp:86]: Hi @laurent-altr  can you explain please the problem you found

Comment (sebastienVilleneuve) [reply engine/source/output/h3d/h3d_build_cpp/c_h3d_create_nodal_scalar_datatype.cpp:86]: Hi @MahaHachani , you need after an array malloc to make somewhere a free(array) in order not to have memory leaks .

**Pull request closed without merging**